git_wrapper/base.py

```diff
@@ -9,20 +9,24 @@
 from git_wrapper import exceptions
 
 
-logger = logging.getLogger(__name__)
-
-
 class GitWrapperBase(object):
     """Provides a wrapper to interact with a git repository"""
 
-    def __init__(self, path='', repo=None):
+    def __init__(self, path='', repo=None, logger=None):
         """Constructor for GitUtilBase object
 
             :param str path: Path to a git repo Default('')
             :param git.Repo repo: An already constructed git.Repo object to use Default(None)
         """
         self.__repo = None  # Added to clear pylint warnings
         self.__setup(path, repo)
+        self._setup_logger(logger)
+
+    def _setup_logger(self, logger):
+        if logger:
+            self.logger = logger
+        else:
+            self.logger = logging.getLogger('git_wrapper')
```

Comment:
Could we still use __name__ here, so the logs are attached to the calling object?

Reply:
Sure. Everything seemed to get associated with "git_wrapper.base" no matter what in my local tests, but I suppose that would be adjusted via a logging config anyway.